client/asset/btc/btc.go

```diff
@@ -495,11 +495,14 @@ func (btc *ExchangeWallet) feeRateWithFallback() uint64 {
 // FundOrder selects utxos (as asset.Coin) for use in an order. Any Coins
 // returned will be locked. Part of the asset.Wallet interface.
 func (btc *ExchangeWallet) FundOrder(value uint64, nfo *dex.Asset) (asset.Coins, error) {
+	btc.log.Debugf("Attempting to fund order for %d based units of %s", value, nfo.Symbol)
 	if value == 0 {
 		return nil, fmt.Errorf("cannot fund value = 0")
 	}
 	// Now that we allow funding with 0 conf UTXOs, some more logic could be
 	// used out of caution, including preference for >0 confs.
+	btc.fundingMtx.Lock()
+	defer btc.fundingMtx.Unlock() // stay locked until we update the map at the end
```

Comment:
also until lockunspent(true, ...) at the end so that spendableUTXOs in another thread doesn't pick coins that are about to be locked here.